Post new participant

EvanDungate opened the pull request:
This is the bare bones functionality to get users from hubspot and post them to saasquatch, I have added comments for some of the work that needs to be done still. I know there are many flaws so all feedback is welcome :)

davidbishoppp committed:
get auth0 JWT from saasquatch

EvanDungate committed:
Update index.ts

adding functionality to get participants list from saasquatch and post new users

EvanDungate committed:
got participants

created new route for integration and now are able to get participants and contacts list as json

EvanDungate committed:
Draft of post participants

This is a draft of reading all hubspot contacts and posting them to saasquatch. This functionality needs to be integrated with Oauth as user creation cannot be done using just an API key. Furthermore, this code needs comments, gherkin and to not look as ugly as it does.

EvanDungate committed:
can now post users to saasquatch

hitting the endpoint localhost:8000/contacts will now create new users in saasquatch based on their firstname lastname and email

EvanDungate committed:
Merge branch 'main' into post-new-participant

EvanDungate committed:
merged main into my branch

The code I am pushing in index.ts is ready to demo but needs more comments and testing

EvanDungate committed:
Update index.ts

Added comments and 500 error for participants no longer exists

MAsimSENG commented on src/routes/index.ts:
```diff
@@ -42,4 +129,47 @@ router.get('/contacts', async (req, res) => {
+
+router.get('/participants', async (req, res) => {
+
+    const Participants = await getParticipants();
```
Perhaps a try-catch here for the await to handle a rejection? 
example scenario for rejection: (which can happen after we are using access tokens for API calls)  user disconnects our integration and access token has expired so the API calls will return an error. In this scenario We must redirect user to authorize the application again. So in the catch we can have res.redirect("/hubspot"))

daniel-czz committed:
create addcontact function

EvanDungate committed:
Merge branch 'main' into post-new-participant

EvanDungate committed:
Update index.ts

addressed muhammads request for a try catch block and resolved merge conflicts with the yarn file

davidbishoppp commented on src/routes/index.ts:
```diff
@@ -42,4 +129,92 @@ router.get('/contacts', async (req, res) => {
+
+
+// add the participants in Saas to Hubspot contact list
+async function addcontacts(participant:any)  {
```
Camel case, sorry for nit pick

EvanDungate replied:
Thanks for catching that, resolved

EvanDungate committed:
Update index.ts

changed posting new participants to saasquatch to use email as their ID and account ID

EvanDungate committed:
ran yarn install

should this file not be included in our merges? I seem to have conflicts with master everytime

EvanDungate committed:
Update index.ts

updated getcontacts to getContacts
